src/UEA/UEA_EVM.sol

```diff
@@ -156,75 +146,147 @@ contract UEA_EVM is ReentrancyGuard, IUEA {
     }
 
     /**
-     * @inheritdoc IUEA
+     * @notice                  Internal handler for multicall execution
+     * @dev                     Executes multiple calls in sequence, reverting if any fails
+     * @dev                     Prevents migration selector in subcalls for safety
+     * @param payload           the UniversalPayload containing multicall data
+     * @return success          whether all calls succeeded
+     * @return returnData       return data from the last call or first failed call
      */
-    function migrateUEA(MigrationPayload calldata payload, bytes calldata signature) external nonReentrant {
-        bytes32 payloadHash = getMigrationPayloadHash(payload);
+    function _handleMulticall(UniversalPayload memory payload) 
+        internal 
+        returns (bool success, bytes memory returnData) 
+    {
+        Multicall[] memory calls = decodeCalls(payload.data);
+
+        for (uint256 i = 0; i < calls.length; i++) {
+            if (isMigration(calls[i].data)) {
+                revert Errors.InvalidCall();
+            }
 
-        if (!verifyPayloadSignature(payloadHash, signature)) {
-            revert Errors.InvalidEVMSignature();
+            (success, returnData) = calls[i].to.call{value: calls[i].value}(calls[i].data);
+            if (!success) {
+                return (success, returnData);
+            }
         }
 
-        unchecked {
-            nonce++;
+        return (true, "");
+    }
+
+    /**
+     * @notice                  Internal handler for migration execution
+     * @dev                     Executes migration via delegatecall to migration contract
+     * @dev                     Enforces safety constraints: must target self, no value transfer
+     * @param payload           the UniversalPayload containing migration data
+     * @return success          whether the migration succeeded
+     * @return returnData       return data from the delegatecall
+     */
+    function _handleMigration(UniversalPayload memory payload) 
+        internal 
+        returns (bool success, bytes memory returnData) 
+    {
+        if (payload.to != address(this)) {
+            revert Errors.InvalidCall();
+        }
+        
+        if (payload.value != 0) {
+            revert Errors.InvalidCall();
         }
 
+        // Decode migration contract address from data
```

Comment:
last time we discussed that the factory contract will be storing the latest migration contract
and in UEA migration, migration contract address would be fetched from factory itself instead of user passing it out